### PR TITLE
impr(cloud-v2#494): Remove OpenSSL as a dependency

### PR DESCRIPTION
**Pull request by indiv0**
### Pull Request Description

**TL;DR:** Replace OpenSSL with rustls so we can compile to musl. The PR also updates my cloud env settings as a minor secondary change.

OpenSSL is hard to compile for two target types:
1. WASM targets
2. statically-compiled (i.e., musl) targets.

Because of the first restriction, OpenSSL is already not used in any of the IDE dependencies directly.

The `cloud-v2` repo now depends on `engine_protocol` crate in order to communicate with the language server. This crate uses OpenSSL during the build process. The `cloud-v2` repo builds statically-compiled binaries. So the OpenSSL dependency makes it impossible to build these binaries because of the second restriction.

Replacing OpenSSL with rustls keeps our ability to communicate over TLS while making it trivial to compile to WASM & musl. Rustls doesn't have the same restrictions as OpenSSL.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]
the only JS changes are the token changes for @indiv0's cloud environment

File: build/deprecated/build-utils/Cargo.toml [hunk 1]
```diff
@@ -11,4 +11,5 @@ serde = { version = "1.0", features = ["derive"] }
 
 [dependencies.reqwest]
```
Comment (indiv0) [inline build/deprecated/build-utils/Cargo.toml]: @mwu-tow it looks like this crate might not even use reqwest anymore? Perhaps this dep (and the others) could be removed entirely?

**Review by PabloBuchu — APPROVED** [review]

File: build/deprecated/build-utils/Cargo.toml (hunk 1 above)
Comment (mwu-tow) [reply build/deprecated/build-utils/Cargo.toml]: @indiv0 I dont think it is unused. I see a number of usages, and this is a transitive dependency of other crates.
Bug you can always try. In general if you can remove something and everything still builds, then go ahead.

File: build/deprecated/build-utils/Cargo.toml [hunk 2]
```diff
@@ -11,4 +11,5 @@ serde = { version = "1.0", features = ["derive"] }
...
 version = "0.11.12"
-features = ["blocking", "json"]
+default-features = false
+features = ["blocking", "json", "rustls-tls"]
```
Comment (mwu-tow) [inline build/deprecated/build-utils/Cargo.toml]: You should be able to get rid of this whole section altogether and use something like 
```
reqwest = { workspace = true }
```
So we keep settings in one place.

**Review by mwu-tow — APPROVED** [review]

File: build/deprecated/build-utils/Cargo.toml (hunk 1 above)
Comment (indiv0) [reply build/deprecated/build-utils/Cargo.toml]: Yeah the deps look unused and everything builds without them; removed.
